vector/src/main/java/im/vector/app/features/voicebroadcast/listening/VoiceBroadcastPlayerImpl.kt

```diff
@@ -263,19 +263,27 @@ class VoiceBroadcastPlayerImpl @Inject constructor(
         if (nextItem != null) {
             isPreparingNextPlayer = true
             sessionScope.launch {
-                prepareMediaPlayer(nextItem.audioEvent.content) { mp ->
-                    isPreparingNextPlayer = false
-                    nextMediaPlayer = mp
-                    when (playingState) {
-                        State.PLAYING,
-                        State.PAUSED -> {
-                            currentMediaPlayer?.setNextMediaPlayer(mp)
-                        }
-                        State.BUFFERING -> {
-                            mp.start()
-                            onNextMediaPlayerStarted(mp)
+                try {
+                    prepareMediaPlayer(nextItem.audioEvent.content) { mp ->
+                        isPreparingNextPlayer = false
+                        nextMediaPlayer = mp
+                        when (playingState) {
+                            State.Playing,
+                            State.Paused -> {
+                                currentMediaPlayer?.setNextMediaPlayer(mp)
+                            }
+                            State.Buffering -> {
+                                mp.start()
+                                onNextMediaPlayerStarted(mp)
+                            }
+                            is State.Error,
+                            State.Idle -> stopPlayer()
                         }
-                        State.IDLE -> stopPlayer()
+                    }
+                } catch (failure: VoiceBroadcastFailure.ListeningError.DownloadError) {
+                    isPreparingNextPlayer = false
+                    if (playingState == State.Buffering || tryOrNull { currentMediaPlayer?.isPlaying } != true) {
```

Comment:
Why do we have this condition to set the state to `Error`? If we catch an Error it means there is an error whatever the current state is, am I correct?

Reply:
In this place, the error is on the next chunk (the current one should be fine). I didn't want to display the error at this moment. When the current chunk will be ended, it will enter in buffering mode and try to prepare again the next chunk, if it fails again, it will show the error.

Reply:
Okay I see. Maybe we could encapsulate and mutualize this condition into a single method with a clear name to understand this intention? If the method name is not enough, we could also add a code comment.

Reply:
I could not find a method name to mutualize the condition, I added a comment instead https://github.com/vector-im/element-android/pull/7919/commits/8a2f28bc376fd2f34c97270cdff7336332efe50e